### PR TITLE
sqlcmd: init at 0.15.3

### PR DESCRIPTION
**Pull request by ratsclub**
###### Description of changes

Follow up of #[221699](https://github.com/NixOS/nixpkgs/pull/221699).

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by mrene** [comment]
Result of `nixpkgs-review pr 221858` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>sqlcmd</li>
  </ul>
</details>

File: pkgs/development/tools/database/sqlcmd/default.nix [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  };
+
+  meta = with lib; {
+    description = "A command line tool for working with Microsoft SQL Server, Azure SQL Database, and Azure Synapse";
+    homepage = "https://github.com/microsoft/go-sqlcmd";
+    changelog = "https://github.com/microsoft/go-sqlcmd/releases";
+    license = licenses.mit;
+    maintainers = with maintainers; [ ratsclub ];
+  };
+}
```
Comment (superherointj) [inline pkgs/development/tools/database/sqlcmd/default.nix:55]:
```suggestion
  meta = {
    description = "A command line tool for working with Microsoft SQL Server, Azure SQL Database, and Azure Synapse";
    homepage = "https://github.com/microsoft/go-sqlcmd";
    changelog = "https://github.com/microsoft/go-sqlcmd/releases";
    license = lib.licenses.mit;
    maintainers = [ lib.maintainers.ratsclub ];
  };
}
```

There is a problem when using `with` for statical analysis, so as a rule of thumb, avoid `with`, specially when nested (like `with lib; with maintainers;`
  - More information: https://github.com/atorres1985-contrib/nixpkgs/commit/451dfa6d84586b5cb83e6a7254689dff66b4523a

File: pkgs/development/tools/database/sqlcmd/default.nix [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+  ldflags = [ "-s" "-w" "-X main.version=${version}" ];
+
+  subPackages = [ "cmd/modern" ];
+
+  preCheck = ''
+    export HOME=$(mktemp -d)
+  '';
+
+  nativeBuildInputs = [ installShellFiles ];
```
Comment (superherointj) [inline pkgs/development/tools/database/sqlcmd/default.nix:31]:
```suggestion
  nativeBuildInputs = [ installShellFiles ];
```
`nativeBuildInputs` is buildPhase. Should be after configurePhase (subPackages) and before checkPhase (preCheck). If you don't want to change this I'll merge anyway.

File: pkgs/development/tools/database/sqlcmd/default.nix [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  passthru.tests.version = testers.testVersion {
+    package = sqlcmd;
+    command = "sqlcmd --version";
+    inherit version;
+  };
+
+  meta = {
+    description = "A command line tool for working with Microsoft SQL Server, Azure SQL Database, and Azure Synapse";
+    homepage = "https://github.com/microsoft/go-sqlcmd";
+    changelog = "https://github.com/microsoft/go-sqlcmd/releases";
```
Comment (superherointj) [inline pkgs/development/tools/database/sqlcmd/default.nix:51]:
```suggestion
    changelog = "https://github.com/microsoft/go-sqlcmd/releases/tag/v${version}";
```
I know... This is getting tiring. But it is almost perfect. lol

**Comment by superherointj** [comment]
Result of `nixpkgs-review pr 221858` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>sqlcmd</li>
  </ul>
</details>

**Review by superherointj — APPROVED** [review]

File: pkgs/development/tools/database/sqlcmd/default.nix [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+  passthru.tests.version = testers.testVersion {
+    package = sqlcmd;
+    command = "sqlcmd --version";
+    inherit version;
+  };
+
+  meta = {
+    description = "A command line tool for working with Microsoft SQL Server, Azure SQL Database, and Azure Synapse";
+    homepage = "https://github.com/microsoft/go-sqlcmd";
+    changelog = "https://github.com/microsoft/go-sqlcmd/releases/tag/v${version}";
```
Comment (superherointj) [inline pkgs/development/tools/database/sqlcmd/default.nix:51]:
```suggestion
    changelog = "https://github.com/microsoft/go-sqlcmd/releases/tags/v${version}";
```

**Pull request merged**